model/lifecycle.go

```diff
@@ -395,22 +411,23 @@ func RestartVersion(versionId string, taskIds []string, abortInProgress bool, ca
 	}
 
 	for tg, t := range taskGroupsToCheck {
-		if err = checkResetSingleHostTaskGroup(&t, caller); err != nil {
+		if err := checkResetSingleHostTaskGroup(&t, caller); err != nil {
 			return errors.Wrapf(err, "resetting task group '%s' for build '%s'", tg.TaskGroup, tg.Build)
 		}
 	}
 
 	// Set all the task fields to indicate restarted
-	if err = MarkTasksReset(restartIds); err != nil {
+	if err := MarkTasksReset(restartIds); err != nil {
 		return errors.WithStack(err)
 	}
-	for _, t := range tasksToRestart {
+	for _, t := range allFinishedTasks {
 		if !t.IsPartOfSingleHostTaskGroup() { // this will be logged separately if task group is restarted
 			event.LogTaskRestarted(t.Id, t.Execution, caller)
 		}
 	}
-	if err = build.SetBuildStartedForTasks(tasksToRestart, caller); err != nil {
-		return errors.Wrap(err, "setting builds started")
+
+	if err := errors.Wrap(build.SetBuildStartedForTasks(allFinishedTasks, caller), "setting builds started"); err != nil {
```

Comment:
Why did we move the wrap up to the if statement level? This is pretty unconventional / harder to read than just keeping the wrap in the next line

Reply:
I am not sure haha - reverted.